### PR TITLE
all: clean up tech debt left behind by the API split

### LINKED ISSUE #2760: Tech debt: remove the full/light split accman service dependency
The full/light node API split PR https://github.com/ethereum/go-ethereum/pull/2159 separated out account/key management to its own node service. This is not a good decision because it makes setting up Ethereum nodes from client libraries much more complex and unwarranted.

The account manager service needs to be moved back cleanly into the Ethereum services and not stand as a standalone service.

### LINKED ISSUE #2806: abigen doesn't support imports
Because abigen invokes the solidity compiler by passing a string, the compiler has no path to work with for imports. It'd be really useful if it supported imports relative to the directory of the root contract being compiled.

### PR DESCRIPTION
**Pull request by fjl**
This PR resolves some of the tech debt created by #2159.
Please see the commit descriptions for more detail about the individual changes.

Fixes #2760 
Fixes #2806, accidentally

### CODE REVIEW

**Comment by robotally** [comment]
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Mon Aug 15 18:21:23 UTC 2016_

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -722,54 +711,29 @@ func MakeSystemNode(name, version string, relconf release.Config, extra []byte,
...
-	}
 	if err := stack.Register(func(ctx *node.ServiceContext) (node.Service, error) {
 		return release.NewReleaseService(ctx, relconf)
 	}); err != nil {
 		Fatalf("Failed to register the Geth release oracle service: %v", err)
 	}
-	return stack
+}
+
+func RegisterShhService(stack *node.Node) {
```
Comment (karalabe) [inline cmd/utils/flags.go]: Please add a doc :)

File: node/node.go [hunk 2]
```diff
@@ -48,6 +50,7 @@ var (
 type Node struct {
 	datadir  string         // Path to the currently used data directory
 	eventmux *event.TypeMux // Event multiplexer used between the services of a stack
+	accman   *accounts.Manager
```
Comment (karalabe) [inline node/node.go]: Please add a doc :P All other fields have one :)

File: node/node.go [hunk 3]
```diff
@@ -128,6 +137,41 @@ func New(conf *Config) (*Node, error) {
 	}, nil
 }
 
+func makeAccountManager(conf *Config) (*accounts.Manager, error) {
```
Comment (karalabe) [inline node/node.go]: Could you please move this method to `node/config.go` and make sure it meshes nicely with the other methods there? All these configuration dependent subtleties are done there.

Comment (karalabe) [inline node/node.go]: i.e. make it 

`func (c *Config) AccountManager()  (*accounts.Manager, error)`

File: node/node.go [hunk 4]
```diff
@@ -128,6 +137,41 @@ func New(conf *Config) (*Node, error) {
...
+		keydir, err = filepath.Abs(conf.KeyStoreDir)
+	default:
+		// There is no datadir.
+		keydir, err = ioutil.TempDir("", "go-ethereum-keystore")
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	if err := os.MkdirAll(keydir, 0744); err != nil {
```
Comment (karalabe) [inline node/node.go]: Perhaps this should be `0600`?

**Comment by karalabe** [comment]
When cleaning up the dapp database, please also clean it out from teh `ethdb` package.

**Comment by karalabe** [comment]
Finished reviewing, apart from the minor issues above, looks very nice!

File: node/node.go (hunk 2 above)
Comment (fjl) [inline node/node.go]: What should the comment say? I seriously don't know what to write about this field. It's kind of self-explanatory that this is the account manager.

File: node/node.go (hunk 4 above)
Comment (fjl) [inline node/node.go]: Yes, good catch.

**Comment by fjl** [comment]
@karalabe PTAL

**Comment by karalabe** [comment]
LGTM :+1:

**Comment by fjl** [comment]
squashed

**Comment by fjl** [comment]
Tests were failing because of the last minute change to keystore permissions. The correct permissions are `0700`, updated now.

**Review by Thompson1985 — APPROVED** [review]

**Review by Thompson1985 — APPROVED** [review]

**Review by Thompson1985 — APPROVED** [review]

**Review by Thompson1985 — APPROVED** [review]

**Pull request merged**